src/App.tsx

```diff
@@ -1,14 +1,67 @@
-/* eslint-disable max-len */
-import React from 'react';
+import { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { FilterType } from './types/FilterType';
+
+const filteredTodos = (
+  query: string,
+  todos: Todo[],
+  filterType: FilterType,
+) => {
+  const propLower = (prop: string) => prop.toLocaleLowerCase();
+
+  const visibleTodos = todos.filter(todo => (
+    propLower(todo.title).includes(propLower(query))
+  ));
+
+  switch (filterType) {
+    case FilterType.ACTIVE:
+      return visibleTodos.filter(todo => !todo.completed);
+
+    case FilterType.COMPLETED:
+      return visibleTodos.filter(todo => todo.completed);
+
+    default:
+      return visibleTodos;
+  }
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+  const [isLoadedUser, setIsLoadedUser] = useState(false);
+  const [isLoadedTodos, setIsLoadedTodos] = useState(false);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState<FilterType>(FilterType.ALL);
+
+  const getTodo = (todoId: number) => {
+    return todos.find(todo => todo.id === todoId) || todos[0];
+  };
```

Comment:
would be nice to move this function outside  component and pass todos here, or use useCallback hook to avoid creating of this function on every render